nan.h

```diff
@@ -801,44 +801,83 @@ class TryCatch {
 #endif
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::Function> func
     , int argc
     , v8::Local<v8::Value>* argv) {
+    v8::Isolate *isolate = v8::Isolate::GetCurrent();
+    v8::EscapableHandleScope scope(isolate);
+    v8::Local<v8::Object> target_obj;
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
-    return New(node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv));
+    if (target->IsUndefined() || target->IsNull()) {
+      target_obj = isolate->GetCurrentContext()->Global();
```

Comment:
@bnoordhuis  Dealing with exceptions for every function call sounds like it would be quite expensive, no?

Here's the same line for `Function::Call` in V8 from 0.8.28
https://github.com/nodejs/node/blob/v0.8.28-release/deps/v8/src/api.cc#L3639

And here's `Utils::OpenHandle`
https://github.com/nodejs/node/blob/v0.8.28-release/deps/v8/src/api.h#L297-L302
There does not seem to be a type check there... I guess the only way of finding out if that crude hack I suggested works is by testing., right?